### ISSUE TITLE
Provide example on how to use Let's Encrypt

### ISSUE THREAD

**Issue by JasonSwindle** [joshix/caddybox#14]
Howdy,

Now that Let's Encrypt is public, and Caddy can auto-magic use LE, can you provide an example via the README.md?

Thanks,
Jason

**Comment by joshix** [comment]
Definite goal, just have to find a little "20% time" to play with the LE stuff in this context myself.

**Comment by joshix** [comment]
related TODO noted on #13

**Comment by cboettig** [comment]
Thanks, advice on this would be great.  It looks like it should work out of the box given a Caddyfile with an https, non-localhost domain name (https://caddyserver.com/docs/automatic-https), but I get the error:

```
2015/12/11 18:27:06 error creating client: get directory at 'https://acme-v01.api.letsencrypt.org/directory': Get https://acme-v01.api.letsencrypt.org/directory: x509: failed to load system roots and no roots provided
```

The docs mention that this approach will put the generated keys in your home directory & that you should be sure to back those up; which would require some extra care in the docker context since that home directory is ephemeral as the container if not linked as a volume...  Not sure if that's connected to my problem.

**Comment by joshix** [comment]
@cboettig Take a look at #6. The caddybox container doesn't include any SSL roots, so until caddybox changes that, you'll need to add them with a volume or by building `FROM` caddybox as a base image.

You'll see that in #7 and #8 we once vendored some root CA's with caddybox, but removed them a few months ago. The LE support is now making me revisit [that decision](https://github.com/joshix/caddybox/pull/8#issuecomment-124929442).

**Comment by cboettig** [comment]
Thanks for the clear explanation, now it makes perfect sense.  

Yup, might be convenient to include the root CAs, but otherwise simply documenting an LE example showing the volume linking to a root CA (I suppose that would be distro-dependent, but could document a/some common ones) would be sufficient.

**Comment by joshix** [comment]
@JasonSwindle @cboettig WIP LE support/docs in #16. What do you think?

There is a next step of some sort where we restore/re-use the generated cert/key pairs for subsequent new `docker runs`. Currently I only cover where they are written and how to grab them.

**Comment by cboettig** [comment]
Nice, added some thoughts over in #16

**Comment by joshix** [comment]
Closed by #16
